taxi/tests/test_views.py

```diff
@@ -0,0 +1,156 @@
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PublicCarTest(TestCase):
+    def test_login_required_car_list_view(self) -> None:
+        response = self.client.get(reverse("taxi:car-list"))
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_car_detail_view(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:car-detail",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_car_create_view(self) -> None:
+        response = self.client.get(reverse("taxi:car-create"))
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_car_update_view(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:car-update",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_car_delete_view(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:car-delete",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_toggle_assign_to_car(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:toggle-car-assign",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PublicDriverTest(TestCase):
+    def test_login_required_driver_list_view(self) -> None:
+        response = self.client.get(reverse("taxi:driver-list"))
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_driver_detail_view(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:driver-detail",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_driver_create_view(self) -> None:
+        response = self.client.get(reverse("taxi:driver-create"))
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_driver_update_view(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:driver-update",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+    def test_login_required_driver_delete_view(self) -> None:
+        response = self.client.get(reverse(
+            "taxi:driver-delete",
+            kwargs={"pk": 1})
+        )
+        self.assertNotEqual(response.status_code, 200)
+
+
+class PrivateTest(TestCase):
+    def setUp(self) -> None:
+        self.user = get_user_model().objects.create_user(
+            username="Eren",
+            password="12345",
+        )
+        self.client.force_login(self.user)
+
+    def test_retrieve_manufacturer_list_view(self) -> None:
+        Manufacturer.objects.create(name="BMW", country="Germany")

```

Comment:
Place objects creation in the `setUp` to not duplicate this logic in different methods